lib/linguist/heuristics.rb

```diff
@@ -226,6 +226,16 @@ def call(data)
       end
     end
 
+    disambiguate ".gml" do |data|
+      if /^\s*(\<\?xml|xmlns)/i.match(data)
+        Language["XML"]
+      elsif /^\s*(graph[\s\n\r]{1,}\[$)|(node[\s\n\r]{1,}\[$)/i.match(data)
```

Comment:
`{1,}` can be written `+`, which is more widely supported by regex engines (Oniguruma, PCRE), even though Ruby uses Oniguruma anyway. We can also simplify to `^\s*(graph|node)[\s\n\r]+\[$`.

Reply:
Thanks! It was quite a lame regex indeed. Updated :)